### PR TITLE
Remove `npm` dependency

### PR DESCRIPTION
**Pull request by kevva**
It's not pretty, but it does the job to get it 99% right.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,5 +1,49 @@
...
+		conf.addFile(projectConf, 'project');
+	} else {
+		conf.add({}, 'project');
+	}
+
+	conf.addFile(conf.get('userconfig'), 'user');
+
+	if (conf.get('prefix')) {
+		const etc = path.resolve(conf.get('prefix'), 'etc');
+		makeDir.sync(etc);
```
Comment (kevva) [inline index.js]: Do we wan't to create these directories or should we just skip? I guess `npm` does it to just *work* but we don't have to.

File: lib/conf.js [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+	add(data, marker) {
+		try {
+			for (const x of Object.keys(data)) {
+				data[x] = util.parseField(data[x], x);
+			}
+		} catch (err) {
+			throw err;
+		}
+
+		return ConfigChain.prototype.add.call(this, data, marker);
```
Comment (sindresorhus) [inline lib/conf.js]:
```js
return super.add(data, marker);
```

File: lib/defaults.js [hunk 3]
```diff
@@ -0,0 +1,152 @@
+
```
Comment (sindresorhus) [inline lib/defaults.js]: Add a comment about this being a generated file.

The indentation is also off. Not important though.

File: lib/defaults.js [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+      shrinkwrap: true,
+      'sign-git-tag': false,
+      'sso-poll-frequency': 500,
+      'sso-type': 'oauth',
+      'strict-ssl': true,
+      tag: 'latest',
+      'tag-version-prefix': 'v',
+      timing: false,
+      tmp: temp,
+      unicode: hasUnicode(),
```
Comment (sindresorhus) [inline lib/defaults.js]: We can just set this to `true` and drop the dependency. It's not important for our use-case.

File: lib/defaults.js [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+      'save-optional': false,
+      'save-prefix': '^',
+      scope: '',
+      'scripts-prepend-node-path': 'warn-only',
+      searchopts: '',
+      searchexclude: null,
+      searchlimit: 20,
+      searchstaleness: 15 * 60,
+      'send-metrics': false,
+      shell: osenv.shell(),
```
Comment (sindresorhus) [inline lib/defaults.js]: Same here. Could just be: 

```
shell: process.platform === 'win32' ? (env.COMSPEC || 'cmd.exe') : (env.SHELL || '/bin/bash')
```

File: lib/defaults.js [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+      tag: 'latest',
+      'tag-version-prefix': 'v',
+      timing: false,
+      tmp: temp,
+      unicode: hasUnicode(),
+      'unsafe-perm': process.platform === 'win32' || process.platform === 'cygwin' || !(process.getuid && process.setuid && process.getgid && process.setgid) || process.getuid() !== 0,
+      usage: false,
+      user: process.platform === 'win32' ? 0 : 'nobody',
+      userconfig: path.resolve(home, '.npmrc'),
+      umask: process.umask ? process.umask() : umask.fromString('022'),
```
Comment (sindresorhus) [inline lib/defaults.js]: We don't care about umask either. Just make it `umask: process.umask()`

File: lib/types.js [hunk 7]
```diff
@@ -0,0 +1,135 @@
+
+	'use strict';
+	const os = require('os');
+	const path = require('path');
+	const Stream = require('stream').Stream;
+	const url = require('url');
+	const semver = require('semver');
```
Comment (sindresorhus) [inline lib/types.js]: We don't need `semver` too. The properties it's needed for is just `npm init`, which is not relevant for us.

File: lib/types.js [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+  'init-author-url': ['', url],
+  'init-license': String,
+  'init-version': semver,
+  json: Boolean,
+  key: [null, String],
+  'legacy-bundling': Boolean,
+  link: Boolean,
+  // local-address must be listed as an IP for a local network interface
+  // must be IPv4 due to node bug
+  'local-address': getLocalAddresses(),
```
Comment (sindresorhus) [inline lib/types.js]: Same here. Not something we need. It's just useful internally for npm.

File: lib/defaults.js [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+	'use strict';
+	const path = require('path');
+	const osenv = require('osenv');
+	const umask = require('umask');
+	const hasUnicode = require('has-unicode');
+
+	const temp = osenv.tmpdir();
+	const uidOrPid = process.getuid ? process.getuid() : process.pid;
+
+	let home = osenv.home();
```
Comment (sindresorhus) [inline lib/defaults.js]: `os.homedir()`

File: lib/defaults.js [hunk 10]
```diff
@@ -0,0 +1,152 @@
+
+	'use strict';
+	const path = require('path');
+	const osenv = require('osenv');
+	const umask = require('umask');
+	const hasUnicode = require('has-unicode');
+
+	const temp = osenv.tmpdir();
```
Comment (sindresorhus) [inline lib/defaults.js]: `os.tmpdir()`

File: lib/defaults.js (hunk 5 above)
Comment (kevva) [inline lib/defaults.js]: How should we maintain this then? Could just skip the `prepublish` script and have this file hard coded or we have to change these values when generating the file. Not to keen on doing it manually everytime I want to publish.

File: index.js (hunk 1 above)
Comment (sindresorhus) [inline index.js]: Skip

File: lib/defaults.js (hunk 5 above)
Comment (sindresorhus) [inline lib/defaults.js]: Yeah, or a simple regex replace, but we're indeed starting to do a lot of modifications.

Comment (kevva) [inline lib/defaults.js]: I think just traversing the object with the parser and comparing it to another object we've defined with the values we want to replace is the easiest option, and also the best.

Comment (sindresorhus) [inline lib/defaults.js]: Agreed

Comment (kevva) [inline lib/defaults.js]: I'm just going to shim the dependencies instead. Easier solution.

**Comment by kevva** [comment]
Removed the unnecessary dependencies and shimmed them instead. I think this should be good to go now.

**Pull request merged**